packages/e2e-tests/specs/experiments/template-part.test.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * WordPress dependencies
+ */
+import { visitAdminPage } from '@wordpress/e2e-test-utils';
+import { addQueryArgs } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import {
+	enableExperimentalFeatures,
+	disableExperimentalFeatures,
+} from '../../experimental-features';
+import { trashExistingPosts } from '../../config/setup-test-framework';
+
+describe( 'Template Part', () => {
+	beforeAll( async () => {
+		await enableExperimentalFeatures( [
+			'#gutenberg-full-site-editing',
+			'#gutenberg-full-site-editing-demo',
+		] );
+		await trashExistingPosts( 'wp_template' );
+		await trashExistingPosts( 'wp_template_part' );
+	} );
+	afterAll( async () => {
+		await trashExistingPosts( 'wp_template' );
+		await trashExistingPosts( 'wp_template_part' );
+		await disableExperimentalFeatures( [
+			'#gutenberg-full-site-editing',
+			'#gutenberg-full-site-editing-demo',
+		] );
+	} );
+	beforeEach( () =>
+		visitAdminPage(
+			'admin.php',
+			addQueryArgs( '', {
+				page: 'gutenberg-edit-site',
+			} ).slice( 1 )
+		)
+	);
+
+	it( 'Should load customizations when in a template even if only the slug and theme attributes are set.', async () => {
+		// Switch to editing the header template part.
+		await page.click(
+			'.components-dropdown-menu__toggle[aria-label="Switch Template"]'
+		);
+		const [ switchToHeaderTemplatePartButton ] = await page.$x(
+			'//button[contains(text(), "header")]'
+		);
+		await switchToHeaderTemplatePartButton.click();
+
+		// Edit it.
+		await page.click( '*[data-type="core/paragraph"]' );
+		await page.keyboard.type( '123' );
```

Comment:
If I understand how this is being asserted against later in the test, it feels pretty risky to rely on the fact that the click occurs somewhere within the paragraph that typing would _append_ to the existing text. If I recall correctly, the `page.click` implementation clicks in the middle of the element dimensions, which may work correctly now, but is definitely fragile and could break suddenly if we do something like change the default viewport size, or default styles of the block list. It would be good if there was some way we could guarantee this is typing where we expect it to be in the selection (avoiding click, or using click with dimensions "to the right" of the element, or using click and then something like <kbd>End</kbd> key or <kbd>Cmd+ArrowRight</kbd> to move to the end of the field).

Reply:
Good catch!

I've updated it to use <kbd>Cmd + ArrowRight</kbd>.

Reply:
Will be interested to see if it works in Travis, since it's a Linux environment, not macOS. Might behave differently.